### DISCUSSION TITLE
Ruby VM memory usage metrics

### DISCUSSION THREAD

**Discussion by dan-corneanu** [open-telemetry/opentelemetry-ruby-contrib#575] [Q&A] [answered]
Hello,
I have set up a docker swarm solution in which I have 
- a rails application instrumented with OpenTelemetry
- otel-collector
- Jaeger to capture and store traces
- Prometheus to capture and store metrics
- Grafana to visualise the metrics and query the traces

I am very happy with the solution so far. I have  a dashboard that shows me the request latency, request rate per minute and error rate per minute. I have also set up the `hostmetrics` receiver on the collector so I also get information about the memory usage on each of my nodes.

However, what I am missing is memory usage statistics for the individual Ruby VMs.

Do you have any suggestions on what I could use to generate per VM memory usage metrics?

**Comment by arielvalentin** [comment, marked as answer]
@dan-corneanu I assume you are looking for metrics like Old/New Gen sizes, GC, Object allocations, etc... 

Unfortunately, the Metrics SDK is still a WIP so you will likely need a vendor specific solution to capture these runtime metrics.

**Reply by dan-corneanu** [reply]
Yes, that is correct. I was able to get memory metrics for Ruby processes using the `hostmetrics` receiver. That works well but I thought it would be nice to also get detailed metrics about the Ruby VM.
